### ISSUE TITLE
Warning for ungrouped data.frame in nest()

### ISSUE THREAD

**Issue by danilinares** [tidyverse/dplyr#5318]
I like the new nest/rowwise approach in dplyr. I am trying to write a function that performs some operation on each group of a data frame or in the whole data fame when it is ungrouped. Something like this:

```r
library(tidyverse)

mtcars_cyl <- mtcars %>% group_by(cyl)

lin_model_fun <- function(.data) {
  .data %>% 
    nest() %>% 
    rowwise() %>% 
    mutate(model = list(lm(mpg ~ wt, data = data)))
}

lin_model_fun(mtcars_cyl)
lin_model_fun(mtcars) # this produces a warning
```

but the function produces a warning when the input data.frame in ungrouped. Is there any possibility to avoid the warning?

**Comment by lionel-** [comment]
With a little R programming, breaking your pipeline into multiple pieces, you could use one of these approaches:

* `dplyr::is_grouped_df()` to determine whether you have a grouped data frame. Adjust the `nest()` call accordingly.

* `dplyr::group_vars()` to get the grouping variables. If empty, you'll want to replace these by all the variables, i.e. the `names()` of the data frame. Then you use `nest(data = all_of(my_vars))`.

Please note that https://community.rstudio.com/ is more appropriate for user questions.

**Comment by danilinares** [comment]
I keep writing here because I guess my comment is more about how this warning affects programming with dplyr.  

In this nest/rowwise approach, the step to check whether the data frame is grouped or ungrouped is unavoidable?

I am coding a new version of the package [quickspy](https://cran.r-project.org/web/packages/quickpsy/index.html) that includes many functions that should perform an operation in a whole data frame if the data frame is ungrouped and on each group if the data frame is grouped. I guess this is something very common when programming with dplyr. I was trying to avoid a conditional in all of these functions to check whether the input data frames is grouped or not. The nest() approach that I wrote here worked well, but it has the warning. I did not expect this warning because the computation that nest performs on an ungroup data frame seemed very natural to me, but I guess you have good reasons to display a warning.
